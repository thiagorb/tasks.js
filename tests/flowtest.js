var Task = require("Task");

new Task(function () {
    console.log("a");
}).then(function (flow) {
    flow.suspend();
    console.log("b");
    setTimeout(function () { flow.resume(); }, 10);
}).then(
    Task.Closure(function () {
        var i = 0;
        return Task.For(
            function () { return i < 3; }, 
            function () { i++; },
            Task.Closure(function () {
                var j = 0;
                return Task.For(
                    function () { return j < 3; },
                    function () { j++; },
                    function (flow) {
                        flow.suspend();
                        console.log("i", i, "j", j);
                        setTimeout(function () { flow.resume(); }, 10);
                    }
                )
            }).then(function () {
                console.log("end j");
            })
        )
    })
).then(function () {
    console.log("c");
}).start();
