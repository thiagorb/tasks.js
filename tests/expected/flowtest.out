a
b
i 0 j 0
i 0 j 1
i 0 j 2
end j
i 1 j 0
i 1 j 1
i 1 j 2
end j
i 2 j 0
i 2 j 1
i 2 j 2
end j
c
